### PR TITLE
Docs: Auto-update copyright year

### PR DESCRIPTION
**Pull request by shati-patel**
See internal linked issue. It would be nice if we could avoid updating the date manually every year (e.g. https://github.com/github/codeql/pull/5028), since this is likely to get overlooked... 📆 

I've checked this in the Sphinx build (both locally and in the actions run 🖼️ )

### CODE REVIEW

File: docs/codeql/index.html [hunk 1]
```diff
@@ -349,11 +349,12 @@ <h4 class="mb-3 text-mono text-gray-light text-normal">Company</h4>
                     </li>
                 </ul>
                 <ul class="list-style-none d-flex text-gray">
-                    <li class="mr-3">&copy; 2021 GitHub, Inc.</li>
+                    <li class="mr-3">&copy; 
+                        <script type="text/javascript">document.write(new Date().getFullYear());</script> GitHub, Inc.</li>
                     <li class="mr-3"><a
-                            href="https://docs.github.com/en/free-pro-team@latest/github/site-policy/github-terms-of-service"
```
Comment (shati-patel) [inline docs/codeql/index.html:354]: (unrelated change: this old URL redirects to a URL without `free-pro-team` anyway)

**Review by ethanpalm — APPROVED** [review]
👍🏼  for docs

**Comment by shati-patel** [comment]
Thanks for the review! 😃 

(The script itself was borrowed from another project, so that should be 👍🏽 too)
